src/lp_sensitivity2.jl

```diff
@@ -0,0 +1,428 @@
+        # matrix \ sparse vector fallbacks.
+        j => B_fact \ collect(std_form.A[:, j])
+        for j = 1:length(basis.basic_cols) if basis.basic_cols[j] == false
+    )
+
+    report = SensitivityReport(
+        Dict{ConstraintRef, Tuple{Float64, Float64}}(),
+        Dict{VariableRef, Tuple{Float64, Float64}}(),
+    )
+
+    ###
+    ### Compute RHS sensitivity
+    ###
+
+    # There is an easy case to consider: a constraint is basic, so we can just
+    # take the distance between the value of the constraint and the
+    # corresponding bound. Otherwise, we need to compute a search direction as
+    # in `_compute_rhs_range`. This is just the negative of the search direction
+    # computed above. Moreover, we have to be careful with doubly-bounded
+    # variables, because our computed range doesn't take into account the
+    # inactive bound.
+
+    for (i, con) in enumerate(std_form.constraints)
+        if basis.constraints[i] == MOI.BASIC
+            report.rhs[con] = _basic_range(con, constraint_object(con).set)
+        else
+            report.rhs[con] = _compute_rhs_range(-d[i + n], x_B, l_B, u_B, atol)
+        end
+    end
+    for (i, con) in enumerate(std_form.bounds)
+        con_obj = constraint_object(con)
+        if basis.bounds[i] == MOI.BASIC
+            report.rhs[con] = _basic_range(con, con_obj.set)
+        else
+            col = std_form.columns[con_obj.func]
+            t_lo, t_hi = _compute_rhs_range(-d[col], x_B, l_B, u_B, atol)
+            if basis.bounds[i] == MOI.NONBASIC_AT_UPPER
+                t_lo = max(t_lo, std_form.lower[col] - x[col])
+            elseif basis.bounds[i] == MOI.NONBASIC_AT_LOWER
+                t_hi = min(t_hi, std_form.upper[col] - x[col])
+            end
+            report.rhs[con] = (t_lo, t_hi)
+        end
+    end
+
+    ###
+    ### Compute objective sensitivity
+    ###
+
+    π = Dict{Int, Float64}(
+        i => reduced_cost(var)
+        for (var, i) in std_form.columns if basis.variables[i] != MOI.BASIC
+    )
+    for (i, c) in enumerate(std_form.constraints)
+        if basis.constraints[i] != MOI.BASIC
+            π[n + i] = is_min ? dual(c) : -dual(c)
+        end
+    end
+
+    for (var, i) in std_form.columns
+        if basis.variables[i] == MOI.NONBASIC_AT_LOWER && is_min
+            @assert π[i] > -atol
+            # We are minimizing and variable `i` is nonbasic at lower bound.
+            # (δ⁻, δ⁺) = (-πᵢ, ∞) because increasing the objective coefficient
+            # will only keep it at the bound.
+            report.objective[var] = (-π[i], Inf)
+        elseif basis.variables[i] == MOI.NONBASIC_AT_UPPER && !is_min
+            @assert π[i] > -atol
+            # We are maximizing and variable `i` is nonbasic at upper bound.
+            # (δ⁻, δ⁺) = (-πᵢ, ∞) because increasing the objective coefficient
+            # will only keep it at the bound.
+            report.objective[var] = (-π[i], Inf)
+        elseif (
```

Comment:
I don't think you need the parentheses around this condition?

Reply:
I ran the formatter instead.